pkg/istio-agent/sds-agent.go

```diff
@@ -136,15 +135,10 @@ type AgentConfig struct {
 	LocalXDSAddr string
 }
 
-// NewSDSAgent wraps the logic for a local SDS. It will check if the JWT token required for local SDS is
-// present, and set additional config options for the in-process SDS agent.
-//
-// The JWT token is currently using a pre-defined audience (istio-ca) or it must match the trust domain (WIP).
-// If the JWT token is not present - the local SDS agent can't authenticate.
-//
-// If node agent and JWT are mounted: it indicates user injected a config using hostPath, and will be used.
-//
-func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig, sopts *security.Options) *Agent {
+// NewAgent wraps the logic for a local SDS. It sets config options for the in-process SDS agent.
+// SDS agent uses a credential fetcher to get the workload credential to provision certificate for the workload.
```

Comment:
That's not even a small part of what it does - it can use existing certificate, etc. And the other details you remove are pretty important on understanding how this works - we should add more comments, not remove.